File: solvingProblemAgent/c++/graph/transition.cpp
#ifndef TRANSITION_CPP
#define TRANSITION_CPP

#include "state.cpp"
#include <iostream>
#include <string>
using namespace std;

class state;

class transition{
    private:
        state* next;
        int cost;
    public:
        transition(state* next, int cost){
            this->next = next;
            this->cost = cost;
        }
        int getCost(){
            return this->cost;
        }
        string toString(){
            return "Transition: "+this->next->getCityName()+", cost: "+to_string(this->cost);
        }
};

#endif

File: solvingProblemAgent/java/src/model/graph/Transition.java
package src.model.graph;

public class Transition {
    private State nextCity;
    private int cost;

    public Transition(State nextCity, int cost){
        this.nextCity = nextCity;
        this.cost = cost;
    }
    public State getNextCity(){
        return this.nextCity;
    }
    public int getCost(){
        return this.cost;
    }
    public String toString(){
        return "Destination: "+this.nextCity.getCityName()+" - Cost: "+this.cost;
    }
}